lib/std/fs.zig

```diff
@@ -1465,22 +1465,91 @@ pub const Dir = struct {
         }
     }
 
+    /// Calls openDirWithAccessAndCreationW recursively to make an entire path 
+    /// (i.e. falling back if the parent directory does not exist). Opens the dir if the path
+    /// already exists and is a directory.
+    /// This function is not atomic, and if it returns an error, the file system may
+    /// have been modified regardless.
+    fn openDirAccessMaskW(self: Dir, sub_path: []const u8, access_mask: u32, no_follow: bool) OpenError!Dir {
```

Comment:
In light of the doc comments, I think a better name for this function would be something like `makeOpenPathAccessMaskW`.